template.json

```diff
@@ -40,7 +41,8 @@
       "lint:eslint": "eslint ./src --ext .js,.jsx,.ts,.tsx --cache",
       "fix": "npm-run-all fix:*",
       "fix:eslint": "npm run lint:eslint -- --fix",
-      "fix:prettier": "prettier \"src/**/*.js,jsx,ts,tsx,scss}\" --write"
+      "fix:prettier": "prettier \"src/**/*.js,jsx,ts,tsx,scss}\" --write",
+      "deploy": "cross-env rsync -avPx build/* ${npm_package_name}@<host>:~/public/ --delete-after"
```

Comment:
For a project I made a version that uploads the build to a folder of the name of the branch. Then you can have multiple versions at the same time:

```
"upload": "cross-var rsync -avPx dist/* '$npm_package_name@<host>:~/public/$FOLDER/' --delete-after",
    "open": "cross-var open-cli https://$npm_package_name.<host>/$FOLDER/",
    "deploy": "cross-env node ./build-tools/script/deploy.js"
```

The content of `./build-tools/script/deploy.js`:
```
const branchName = require('current-git-branch');
const {exec} = require('child_process');


async function execute(command) {
  console.log(command);

  return new Promise((resolve, reject) => {
    exec(command, (error, output, x) => {
      console.log(x);
      if (error) {
        reject(error);
      } else {
        resolve(output);
      }
    });
  });
}

async function start() {
  const branch = branchName();
  const folder = branch.replace('/', '-');

  console.log(`Building branch '${branch}' for Monkapps`);

  await execute(`yarn build --publicPath=${folder}`);
  await execute(`cross-env-shell FOLDER=${folder} yarn upload`);
  await execute(`cross-env-shell FOLDER=${folder} yarn open`);

  console.log(`Done!`);

}

start();
```

Reply:
Seems useful, especially the ability to defined the deploy folder. Though, I'd replace `current-git-branch` with a command call to `git branch --show-current`. Just so we wouldn't need to add a one use dependency.

Reply:
I added a proper deploy script, by default it's going to just build the project and deploy it. If you pass `--for-branch` to it, then it will build & deploy the project for the current branch.